aten/src/ATen/native/cpu/ReduceOpsKernel.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "vec256.h"
+#include "parallel.h"
+
+#include "ATen/ATen.h"
+#include "ATen/ExpandUtils.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/WrapDimUtils.h"
+
+#include <algorithm>
+#include <functional>
+#include <numeric>
+#include <vector>
+
+#include <iostream>
+#include <map>
+
+
+namespace at {
+namespace native {
+
+using sumImplF = void(Tensor &, const Tensor &, size_t, bool);
+sumImplF sumImplAVX;
+sumImplF sumImplAVX2;
+sumImplF sumImplNONE;
```

Comment:
I'm starting to wonder if this won't cause us missing symbols if you'll be on a machine that doesn't support AVX2 - these functions are only compiled if the current CPU supports their capabilities. We might want to switch them to be always built (even if the current machine can't run them). Also, if we decide to do this, it will probably be clearer to use the specific capability as an enum template argument (both to implementation functions, and as an extra template arg to `vec256`).

Reply:
Good point,but won't this be achieved by removing the arch detection code within the cmake file (without any code changes)?

Reply:
If we use march=native for all compilation units the instructions used will depend on [what cpu is used to compile pytorch](https://wiki.gentoo.org/wiki/GCC_optimization#-march). This will then cause all copies to be compiled with avx2 if the current cpu supports it.

Reply:
Wouldn't this be fixed by enforcing appropriate arch flags for every instantiation of this file?

Reply:
That's what we do, it's part of the {C_AVX2_FLAGS} variable, which comes from THVector. Maybe we should just hardcode that.